src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -1933,6 +1933,140 @@ public Course addCourseWithThreeFileUploadExercise() {
         return courseRepoContent.get(0);
     }
 
+    public Course addCourseWithExercisesAndSubmissions(int numberOfExercises, int numberOfSubmissionPerExercise, int numberOfAssessments, int numberOfComplaints,
+            boolean typeComplaint, int numberComplaintResponses, String validModel) {
+        Course course = ModelFactory.generateCourse(null, pastTimestamp, futureFutureTimestamp, new HashSet<>(), "tumuser", "tutor", "instructor");
+        var tutors = userRepo.getTutors(course);
+        for (int i = 0; i < numberOfExercises; i++) {
+            var currentUser = tutors.get(i % 4);
+
+            if ((i % 3 + 0) == 0) {
```

Comment:
Can you maybe explain what the reason for this  modulo stuff here is? Seems a little bit strange...

Reply:
I added a comment explaining this